spec/memoize_query/cached_select/bind_params_spec.rb

```diff
@@ -0,0 +1,91 @@
+RSpec.describe RedisMemo::MemoizeQuery::CachedSelect::BindParams do
+  it 'unions' do
+    left = described_class.new
+    left.params[1] << { a: 1 }
+
+    right = described_class.new
+    right.params[1] << { b: 2 }
+
+    bp = left.union(right)
+    bp.extract!
+    expect(bp.params[1].to_a).to eq([{ a: 1 }, { b: 2 }])
+
+    left.params[1] << { b: 3 }
+    bp.extract!
+    expect(bp.params[1].to_a).to eq([{ a: 1 }, { b: 3 }, { b: 2 }])
+  end
+
+  it 'products' do
+    left = described_class.new
+    left.params[1] << { a: 1 }
+
+    right = described_class.new
+    right.params[1] << { b: 2 }
+
+    bp = left.product(right)
+    bp.extract!
+    expect(bp.params[1].to_a).to eq([{ a: 1, b: 2 }])
+  end
+
+  it 'excludes conflict query conditions' do
```

Comment:
I might have asked this before 😅  what happens to a SQL query if there's a conflicting query condition and it hits the real database? What gets returned?

Reply:
Oh, it just won't get return anything from the database server -- when performing the query, some query analyzer might even just take out the contradicting part automatically before executing the query